lib/connextclient/ConnextClient.ts

```diff
@@ -400,7 +400,17 @@ class ConnextClient extends SwapClient {
   }
 
   public removeInvoice = async (rHash: string) => {
-    this.expectedIncomingTransfers.delete(rHash);
+    const expectedIncomingTransfer = this.expectedIncomingTransfers.get(rHash);
+    if (expectedIncomingTransfer) {
+      // resolve a hashlock with a paymentId but no preimage to cancel it
+      await this.sendRequest('/hashlock-resolve', 'POST', {
+        assetId: expectedIncomingTransfer.tokenAddress,
+        paymentId: rHash,
```

Comment:
This won't work because `paymentId` is `hash(preimage + assetId)`. I tested https://github.com/connext/rest-api-client/pull/60 manually using the paymentId and the receiver was able to cancel the transfer.

Reply:
Thanks, this wasn't completely clear to me. Do you happen to have an example or instructions of how exactly this hash is calculated? Do I append the token address to the preimage, decode from hex to bytes, and then sha256 the bytes?

Reply:
As discussed offline I think this would be easier if we could get the `paymentId` from the response of `/hashlock-status/$HASH/$ASSET_ID`

Reply:
I implemented a solution to save the `paymentId` from the created event. It won't persist through an xud restart but it'll allow us to cancel invoices for swaps where xud stays online (which should be most cases).